provider/ethereum/token/client.go

```diff
@@ -279,9 +302,34 @@ func (c *client) lookupNFT(ctx context.Context, chain uint64, address common.Add
 		return nil, err
 	}
 
+	// Cache the token metadata to Redis.
+	if c.rueidisClient != nil {
+		if value, err := json.Marshal(tokenMetadata); err == nil {
+			command := c.rueidisClient.B().Setex().
+				Key(c.buildCacheKey(chainID, address, id)).
+				Seconds(int64(defaultCacheDuration.Seconds())).
+				Value(string(value)).
+				Build()
+
+			c.rueidisClient.Do(ctx, command)
```

Comment:
missing error check

Reply:
repeated logic: see L179-190.
it's better to have an abstract func cacheTokenMetadata